File: QTD.SPA/src/app/_DtoModels/ReportSkeleton/ReportSkeletonFilter.ts
export class ReportSkeletonFilter {
    reportSkeletonId: number;
    display: string;
    propertyType: filterPropertyTypeEnum;
    valueType: filterValueTypeEnum;
    value: string;
    defaultValue: string;
    minOption: Date;
    maxOption: Date;
    filterOption: string;
    name: string;
    maxAllowedSelections?:number;
}

export enum filterValueTypeEnum {
    Single,
    Array,
    Range
}
export enum filterPropertyTypeEnum {
    Date,
    Int,
    Boolean,
    String
}

File: QTD2.Domain/Entities/Core/ReportSkeleton/ReportSkeletonFilter.cs
using QTD2.Domain.Entities.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Domain.Entities.Core
{
    public class ReportSkeletonFilter: Entity
    {
        public int ReportSkeletonId { get; set; }
        public string Name { get; set; }
        public FilterPropertyTypeEnum PropertyType { get; set; }
        public FilterValueTypeEnum ValueType { get; set; }
        public DateTime MinOption { get; set; }
        public DateTime MaxOption { get; set; }
        public string FilterOption { get; set; }
        public string DefaultValue { get; set; }
        public int? MaxAllowedSelections { get; set; }
    }
}
